File: bwl.Imaging/Geometry/Polygon/Line.vb
Imports System.Drawing
Imports System.Runtime.Serialization

<DataContractAttribute()>
Public Class Line
    Inherits Polygon

    <IgnoreDataMember>
    Public Property Point1 As PointF
        Set(value As PointF)
            _points(0) = value
        End Set
        Get
            Return _points(0)
        End Get
    End Property

    <IgnoreDataMember>
    Public Property Point2 As PointF
        Set(value As PointF)
            _points(1) = value
        End Set
        Get
            Return _points(1)
        End Get
    End Property

    Public Sub New()
        MyBase.New(False, New PointF, New PointF)
    End Sub

    Public Sub New(x1 As Single, y1 As Single, x2 As Single, y2 As Single)
        MyBase.New(False, New PointF(x1, y1), New PointF(x2, y2))
    End Sub

    Public Sub New(p1 As PointF, p2 As PointF)
        MyBase.New(False, p1, p2)
    End Sub

    Public Overrides Function ToString() As String
        Return "Line: " + Point1.ToString + " - " + Point2.ToString
    End Function
End Class

File: Bwl.Imaging/Geometry/Polygon/Line.cs
using System.Drawing;
using System.Runtime.Serialization;

namespace Bwl.Imaging
{

    [DataContract()]
    public class Line : Polygon
    {

        [IgnoreDataMember]
        public PointF Point1
        {
            set
            {
                _points[0] = value;
            }
            get
            {
                return _points[0];
            }
        }

        [IgnoreDataMember]
        public PointF Point2
        {
            set
            {
                _points[1] = value;
            }
            get
            {
                return _points[1];
            }
        }

        public Line() : base(false, new PointF(), new PointF())
        {
        }

        public Line(float x1, float y1, float x2, float y2) : base(false, new PointF(x1, y1), new PointF(x2, y2))
        {
        }

        public Line(PointF p1, PointF p2) : base(false, p1, p2)
        {
        }

        public override string ToString()
        {
            return "Line: " + Point1.ToString() + " - " + Point2.ToString();
        }
    }
}